app/controllers/update/RemovePlatformOperatorController.scala

```diff
@@ -53,41 +52,27 @@ class RemovePlatformOperatorController @Inject()(
 
   def onPageLoad(operatorId: String): Action[AnyContent] = (identify andThen getData(Some(operatorId)) andThen requireData) { implicit request =>
     getAnswer(BusinessNamePage) { businessName =>
-
-      val form = formProvider(businessName)
-
-      Ok(view(form, operatorId, businessName))
+      Ok(view(formProvider(businessName), operatorId, businessName))
     }
   }
 
   def onSubmit(operatorId: String): Action[AnyContent] = (identify andThen getData(Some(operatorId)) andThen requireData).async { implicit request =>
-
-    super.hc(request)
-
     getAnswerAsync(BusinessNamePage) { businessName =>
-
-      val form = formProvider(businessName)
-
-      form.bindFromRequest().fold(
-        formWithErrors =>
-          Future.successful(BadRequest(view(formWithErrors, operatorId, businessName))),
-
-        value =>
-          if (value) {
-            for {
-              _                 <- platformConnector.removePlatformOperator(operatorId)
-              cleanedData    = request.userAnswers.copy(data = Json.obj())
-              updatedAnswers    <- Future.fromTry(cleanedData.set(PlatformOperatorDeletedQuery, businessName))
-              _                 <- sessionRepository.set(updatedAnswers)
-              subscriptionInfo  <- subscriptionConnector.getSubscriptionInfo
-              _                 <- emailService.sendEmail(RemovedPlatformOperatorRequest.build(request.userAnswers, subscriptionInfo))
-              _                 <- emailService.sendEmail(RemovedAsPlatformOperatorRequest.build(request.userAnswers))
-              _                 <- auditService.sendAudit[RemovePlatformOperatorAuditEventModel](
-                                  RemovePlatformOperatorAuditEventModel(businessName, operatorId).toAuditModel)
-            } yield Redirect(routes.PlatformOperatorRemovedController.onPageLoad(operatorId))
-          } else {
-            Future.successful(Redirect(routes.PlatformOperatorController.onPageLoad(operatorId)))
-          }
+      formProvider(businessName).bindFromRequest().fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, operatorId, businessName))),
```

Comment:
missing test for formWithErrors.

Reply:
This is a missing test from when the page was implemented initially. I have added a test to cover that now.